src/OneMightyRoar/PHP_Paulus_Components/BasicApp.php

```diff
@@ -155,10 +155,10 @@ public function getEnvironment()
      * @access public
      * @return BasicApp
      */
-    public function prepare($auto_load_routes = null, RouteLoader $route_loader = null)
+    public function prepare($auto_load_routes = true, RouteLoader $route_loader = null)
     {
         // Change our defaults
-        $auto_load_routes = (null === $auto_load_routes) ? true : false;
+        $auto_load_routes = (null === $auto_load_routes) ? true : $auto_load_routes;
```

Comment:
PHP doesn't have strict type hints on scalars. It also doesn't have a very good way of "ignoring" arguments with default values, which means that devs have to look at the method definitions themselves (in code) to find out what the default values are for most arguments.

... So a common way of allowing default values to be passed through is to just handle the null case, until we have something like this of course:
https://wiki.php.net/rfc/skipparams